lib/segment/analytics/request.rb

```diff
@@ -55,7 +55,7 @@ def post(write_key, batch)
             body = JSON.parse(res.body)
             error = body["error"]
           end
-        rescue Exception => e
+        rescue JSON::ParserError, Net::ReadTimeout, Net::OpenTimeout, EOFError => e
```

Comment:
I think that it'd be better to rescue `StandardError`, as there are other potential exceptions that could slip through the cracks here (`SocketError`, for example). 

[This StackOverflow question](https://stackoverflow.com/questions/5370697/what-s-the-best-way-to-handle-exceptions-from-nethttp) has a few good answers relating to the subject.

Reply:
Note: I've implemented this in #141